overlord/snapstate/snapmgr.go

```diff
@@ -147,6 +154,46 @@ func (snapst *SnapState) CurrentSideInfo() *snap.SideInfo {
 	panic("cannot find snapst.Current in the snapst.Sequence")
 }
 
+func (snapst *SnapState) PreviousSideInfo() *snap.SideInfo {
+	n := len(snapst.Sequence)
+	if n < 2 {
+		return nil
+	}
+	// find "current" and return the one before that
+	currentIndex := snapst.currentIndex()
+	if currentIndex == 0 {
+		return nil
+	}
+	return snapst.Sequence[currentIndex-1]
+}
+
+func (snapst *SnapState) findIndex(rev snap.Revision) int {
+	for i, si := range snapst.Sequence {
+		if si.Revision == rev {
+			return i
+		}
+	}
+	return -1
+}
+
+func (snapst *SnapState) currentIndex() int {
+	return snapst.findIndex(snapst.Current)
+}
+
+func (snapst *SnapState) Block() []snap.Revision {
+	// return revisions from Sequence[currentIndex:]
+	out := []snap.Revision{}
+	currentIndex := snapst.currentIndex()
+	// panic?
```

Comment:
No need to panic (unset should also return an empty list without it being a problem).

out sould be moved after the conditional, and might become:

out := make([]snap.Revision, 0, len(snapst.Sequence))

(no need to allocate a zero-length array)